api/src/main/java/org/openmrs/module/muzima/handler/JsonRegistrationQueueDataHandler.java

```diff
@@ -275,35 +272,81 @@ private String getPatientUuidFromPayload(){
     }
 
     private void setPatientAddressesFromPayload(){
-        PersonAddress patientAddress = new PersonAddress();
-        Object patientAddressObject = JsonUtils.readAsObject(payload,"$['patient']['patient.personaddress']");
-        if (JsonUtils.isPathAJSONArray(patientAddressObject)){
+
+        Object patientAddressObject = JsonUtils.readAsObject(payload,"$['patient.personaddress']");
+        if (JsonUtils.isPathAJSONArray(patientAddressObject).equals(true)){
             //process as JSONArray
-            String county = JsonUtils.readAsString(payload, "$['patient']['patient.county']");
-            patientAddress.setStateProvince(county);
+            PersonAddress firstSetPatientAddress = new PersonAddress();
```

Comment:
If JSONArray, there can be more than two addresses, you will therefore need to amend so as to accommodate a JSONArray of any length

Reply:
I also guessed the payload could contain more than two patient.personaddress objects or nodes, unfortunately, I limited myself to the example payloads in the notes, will make the necessary adjustments.